alphagov/paas-cf
# [#149806558] Deploy the Compose Scraper according to ENABLE_DATADOG

## What

* Deploys the [new Compose scraper](https://github.com/alphagov/paas-compose-scraper/pull/1) onto the platform when `ENABLE_DATADOG=true`.
* Creates relevant Datadog monitors with Terraform if `ENABLE_DATADOG=true`:
  1. Alert if the scraper reports problems scraping (missing data or the scraper not running.)
  2. The objective of this story: alerting if we go past ~80% resource utilisation on any host of the Compose cluster.

## How to review

1. Wait until https://github.com/alphagov/paas-compose-scraper/pull/1 is merged and then ask @46bit to update this branch to use the newly-tagged commit.
2. Run the `upload-compose-secrets` and `upload-datadog-secrets` Make targets, to upload the new credentials the scraper needs.
3. Deploy this branch of paas-cf to your environment with:
   ```
   BRANCH=149806558-compose-metric-alerting-with-datadog-flag \
     SELF_UPDATE_PIPELINE=false \
     ENABLE_DATADOG=true \
     make dev pipelines
   ```
4. Check that these metrics (for 3 hosts) are showing up in Datadog tagged as `deployment:$DEPLOY_ENV`:
   * `compose.cluster.host.ram.in_use` (RAM percentage used)
   * `compose.cluster.host.ram.used` (RAM GB used)
   * `compose.cluster.host.ram.total` (RAM total GB usable)
   * `compose.cluster.host.disk.in_use` (Disk percentage used)
   * `compose.cluster.host.disk.used` (Disk GB used)
   * `compose.cluster.host.disk.total` (Disk total GB usable)
5. Disable the scraper (e.g., stop the app) and check the Datadog monitor alerts on No Data. This has not been tested yet and needs doing in review. Talk to @46bit for details.
6. Break the scraper (e.g., make its URL 404) and check the Datadog monitor alerts on No Data and a Critical warning on the healthcheck.

## Who can review

Not @46bit @timmow @henrytk @combor @chrisfarms
